### PR TITLE
Add etcd-quorum-guard to the machine config operator

### PR DESCRIPTION
**Pull request by RobertKrawitz**
**- What I did**
Add etcd-quorum-guard to the machine config operator

**- How to verify it**
Until the e2e is added, look at the log files in the artifacts.

**- Description for the changelog**
Add guard to prevent more than one etcd member from being taken down at any one time

### CODE REVIEW

File: install/0000_30_machine-config-operator_07_etcdquorumguard_deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+            declare -r croot=/mnt/kube
+            set -x
+            declare -r health_endpoint="https://127.0.0.1:2379/health"
+            declare -r cert="$(find $croot -name 'system:etcd-peer*.crt' -print -quit)"
+            declare -r key="${cert%.crt}.key"
+            declare -r cacert="$croot/ca.crt"
+            ls -lR "$croot"
+            ls -lRL "$croot"
+            while : ; do date; curl --max-time 2 --cert "${cert//:/\:}" --key "$key" --cacert "$cacert" "$health_endpoint"; sleep 5; done
+        readinessProbe:
```
Comment (hexfusion) [inline install/0000_30_machine-config-operator_07_etcdquorumguard_deployment.yaml]: small nit but do you want `--silent` here as well or is that intentional for debug?

Comment (RobertKrawitz) [reply install/0000_30_machine-config-operator_07_etcdquorumguard_deployment.yaml]: This is for debugging.  I'm going to replace that whole thing with sleep infiinity after I've verified that the eqg is running.

**Comment by RobertKrawitz** [comment]
@runcom @hexfusion debug output in logfiles e. g. https://storage.googleapis.com/origin-ci-test/pr-logs/pull/openshift_machine-config-operator/671/pull-ci-openshift-machine-config-operator-master-e2e-aws/3350/artifacts/e2e-aws/pods/openshift-machine-config-operator_etcd-quorum-guard-59df4d975-2rj84_etcd-quorum-guard-container.log.gz indicates that etcd-quorum-guard is starting up and functioning correctly.

**Comment by runcom** [comment]
> @runcom @hexfusion debug output in logfiles e. g. https://storage.googleapis.com/origin-ci-test/pr-logs/pull/openshift_machine-config-operator/671/pull-ci-openshift-machine-config-operator-master-e2e-aws/3350/artifacts/e2e-aws/pods/openshift-machine-config-operator_etcd-quorum-guard-59df4d975-2rj84_etcd-quorum-guard-container.log.gz indicates that etcd-quorum-guard is starting up and functioning correctly.

do you prefer having the e2e in this PR or as a follow-up? (I would strongly have that tho) - also you'll probably need to raise e2e timeout when adding that.

**Comment by RobertKrawitz** [comment]
> > @runcom @hexfusion debug output in logfiles e. g. https://storage.googleapis.com/origin-ci-test/pr-logs/pull/openshift_machine-config-operator/671/pull-ci-openshift-machine-config-operator-master-e2e-aws/3350/artifacts/e2e-aws/pods/openshift-machine-config-operator_etcd-quorum-guard-59df4d975-2rj84_etcd-quorum-guard-container.log.gz indicates that etcd-quorum-guard is starting up and functioning correctly.
> 
> do you prefer having the e2e in this PR or as a follow-up? (I would strongly have that tho) - also you'll probably need to raise e2e timeout when adding that.

follow-up

**Pull request merged**